src/capture/capture_video.cpp

```diff
@@ -355,37 +355,65 @@ static void create_avi_file(const uint16_t width, const uint16_t height,
 	video.audio.bytes_written   = 0;
 }
 
-// Reads the image's pixels from the emulated DOS video RAM (which is always
-// little-endian) and passes them in the same byte-order to the encoder.
+// Performs some transforms on the passed down rendered image to make sure
+// we're capturing the raw output, then passes down the result in the same
+// byte-order to the encoder. Endianness varies per pixel format (see
+// PixelFormat in video.h for details), the ZMBV encoder handles all that
+// detail.
 //
-// Ideally this could just be one memcpy, but many DOS video modes need their
-// pixels either doubled horizontally and/or vertically to produce the correct
-// on-screen size. So extra copies are needed in both these cases.
+// We always write non-double-scanned and non-pixel-doubled frames in raw
+// video capture mode :
 //
-static void compress_frame(const RenderedImage& image)
+// - Pixel-doubling is not a problem as that's always performed as a
+//   post-processing step; it's never "baked-into" the rendered image.
+//   We just need to completely ignore the `double_width` flag.
+//
+// - Double scanning can be either "baked-in" or performed in post-processing.
+//   Ignoring the `double_height` flag takes care of the post-processing
+//   variety, but for "baked-in" double scanning, we need to skip every second
+//   row.
+//
+// So for example the 320x200 13h VGA mode is always written as 320x200 in raw
+// capture mode regardless of the state of the width & height doubling flags,
+// and irrespective of whether the passed down image is 320x200 or 320x400
+// with "baked-in" double scanning.
+//
+// Composite modes are special because they're rendered at 2x the width of the
+// video mode to allow enough vertical resolution for representing composite
+// artifacts (so 320x200 is rendered as 640x200, and 640x200 as 1280x200).
+// These are written as-is, otherwise we'd be losing information.
+//
+static void compress_raw_frame(const RenderedImage& image)
 {
 	const auto& src = image.params;
-	auto src_row = image.image_data;
+	auto src_row    = image.image_data;
 
-	const auto pixel_multiple = (src.double_width ? 2 : 1);
-	const auto row_multiple   = (src.double_height ? 2 : 1);
+	// We always write non-double-scanned frames in raw video capture mode.
+	// Therefore, to reconstruct the raw image, we must skip every second row
+	// if we're dealing with "baked in" double scanning.
+	const auto num_rows_to_skip = (image.params.rendered_double_scan ? 1 : 0);
```

Comment:
This commit moved the function from capturing "rendered" to "raw", and I think we can lean into what _raw_ means by using that term in the loop variables. 

It will essentially document the raw aspect, and make the code more elegant too.

The raw height (in rows) is when the rendered row multiples are divided out.

We need to pitch across the rendered rows, so the rendered pitch (in width) is when the image.pitch is multiplied by the rendered row multiplies.

This will eliminate all the row_skip `ecrementing` and `continuing` stuff for having the loops operate directly on the raw and rendered values.

Will go from..
```c++
auto row_skip_counter = 0;

const auto can_use_src_directly = (src_bpp == dest_bpp);
if (can_use_src_directly) {
	for (auto i = 0; i < src.height; ++i, src_row += image.pitch) {
		if (row_skip_counter > 0) {
			--row_skip_counter;
			continue;
		}
		compress_row(src_row);
		row_skip_counter = num_rows_to_skip;
	}
	return;
}
```

to:
```c++
const auto can_use_src_directly = (src_bpp == dest_bpp);
if (can_use_src_directly) {
	for (auto i = 0; i < raw_height; ++i, src_row += rendered_pitch) {
		compress_row(src_row);
	}
	return;
}
```

(similarly for the second loop)

Reply:
Done.